cli/server.go

```diff
@@ -1622,30 +1633,29 @@ func startBuiltinPostgres(ctx context.Context, cfg config.Root, logger slog.Logg
 	return connectionURL, ep.Stop, nil
 }
 
-func ConfigureHTTPClient(ctx context.Context, clientCertFile, clientKeyFile string, tlsClientCAFile string) (context.Context, *http.Client, error) {
+func ConfigureHTTPClient(ctx context.Context, clientCertFile, clientKeyFile string) (context.Context, *http.Client, error) {
+	tlsClientConfig := &tls.Config{} //nolint:gosec
+
 	if clientCertFile != "" && clientKeyFile != "" {
 		certificates, err := loadCertificates([]string{clientCertFile}, []string{clientKeyFile})
 		if err != nil {
 			return ctx, nil, err
 		}
+		tlsClientConfig.Certificates = certificates
+		tlsClientConfig.NextProtos = []string{"h2", "http/1.1"}
+	}
 
-		tlsClientConfig := &tls.Config{ //nolint:gosec
-			Certificates: certificates,
-			NextProtos:   []string{"h2", "http/1.1"},
-		}
-		err = configureCAPool(tlsClientCAFile, tlsClientConfig)
-		if err != nil {
-			return nil, nil, err
-		}
+	err := configureRootCAPool(tlsClientConfig)
```

Comment:
Previously we were setting `ClientCAs` on the client which I think is wrong?  Now we only set `tlsConfig.RootCAs`.  But maybe I have misunderstood something and we do need `tlsConfig.ClientCAs` on the client as well.